Library/ENV/4.3/cc

```diff
@@ -1,6 +1,10 @@
-#!/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby -W0
+#!/bin/sh
+# Make sure this shim uses the same Ruby interpreter that is used by Homebrew.
+unset RUBYLIB
+unset RUBYOPT
+exec "$HOMEBREW_RUBY_PATH" -x "$0" "$@"
```

Comment:
This is briefly explained in my commit message:

> The magic basically lies in executing `ruby` with the `-x` option (supported since at least 1.8) and in the following shebang line.

This option tells Ruby to skip everything until it encounters a shebang line that contains `ruby`. That's apparently the standard way to create shell wrappers (Ruby itself uses it in its scripts when built with the `--enable-load-relative` option to get a relocatable build). The main benefit over wrappers in separate files is that the file stays the same, so `$0` is unchanged in the Ruby code.

(I previously experimented with a shell-only `cc` that would invoke a `cc.rb`, but that created a lot of issues and was much more difficult to get right.)